packages/application/src/shell.ts

```diff
@@ -1638,13 +1638,14 @@ export class LabShell extends Widget implements JupyterFrontEnd.IShell {
     args: FocusTracker.IChangedArgs<Widget>
   ): void {
     if (args.newValue) {
-      args.newValue.title.className += ` ${CURRENT_CLASS}`;
+      args.newValue.title.className = `${args.newValue.title.className} ${CURRENT_CLASS} ${ACTIVE_CLASS}`;
+      args.newValue.activate();
+      this._activeChanged.emit(args);
```

Comment:
What about checking if the active element is not in the shell?

```suggestion
      if (!this.node.contains(document.activeElement)) {
        args.newValue.activate();
        this._activeChanged.emit(args);
      }
```

Reply:
Thanks for suggesting something. I tried this and `!this.node.contains(document.activeElement)` is always false-y so new current widget is never activated and focus reverts to `body`. Not sure why this is the case as `this` refers to Lab Shell and `document.activeElement` at that point is still closed tab's content